pkg/config/configuration.go

```diff
@@ -25,6 +25,9 @@ const (
 )
 
 type Configuration struct {
+	// ConfigFile is not an actual configuration variable, but is used to let usages know what file was loaded for the
+	// configuration.
+	ConfigFile string `yaml:"-"`
```

Comment:
Make this a private field, and have it only accessible via a method to make it immutable.